### PR TITLE
add ignore config and read current folder setting

### PR DESCRIPTION
**Pull request by Casear**
I am writing a isomorphic website.
I don't want to restart server if I modify some ui template files.
I add ignore config to .node-dev.json to skip some files.

### CODE REVIEW

**Comment by gasi** [comment]
:+1: To supporting some form of _ignore_ configuration for _universal_ (_isomorphic_) websites, e.g. ones with React.js components that can be server-side rendered but don’t require full server restart upon changes to component files.

**Comment by gasi** [comment]
Thanks for your contribution, @Casear. I added support for ignoring paths with #100 but I see that your implementation is smarter as it makes a distinction between files and directories. Instead of using `indexOf` for file path comparison, I wonder if there is a library out there that does path matching. For example, Babel supports `babel --ignore foo/bar*/*.js` patterns: https://babeljs.io/docs/usage/cli/#options

Would you be interested to investigate? If so, could you please add some test cases to ensure the behavior works as expected. For an example, check out: https://github.com/fgnass/node-dev/commit/95aa2777454686d193a964c9fac2c24dd3c8501f

_P.S. I noticed your PR introduces a lot of whitespace changes. Please check your editor and make sure the PR is minimal and follows existing whitespace conventions. PR without whitespace changes: https://github.com/fgnass/node-dev/pull/98/files?w=1*_

**Comment by Casear** [comment]
@gasi  I add minimatch to do path pattern verify , remove whitespace and add test .I think this version should be ok.

**Comment by Casear** [comment]
@gasi  I check current folder .node-dev.json if there is no .node-dev.json on main.js folder.
I want to do that because I will trigger node-dev on gulp with webpack
```
gulp.task('node-dev',function(){
    nodedev(['src/server.ls'])
})
```
but my node-dev will put the same folder with gulp.js.
That's why I checkout trigger process folder too.

File: lib/cfg.js [hunk 1]
```diff
@@ -13,7 +13,13 @@ function resolvePath(unresolvedPath) {
...
   var dir = main ? path.dirname(main) : '.'
   var c = read(dir)
-
+  if(c==null && dir!='.'){
+    dir = path.dirname('.');
+    c = read(dir);
+  }
+  if(c==null){
+    c={}
+  }
```
Comment (gasi) [inline lib/cfg.js]: Could you add a test case for this scenario? I am not very familiar with how Gulp works. Additionally, please update the docs on how the `.node-dev.json` lookup process works with this change: https://github.com/fgnass/node-dev#settings

Thanks :+1:

File: lib/index.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ var path = require('path')
 var filewatcher = require('filewatcher')
 var ipc = require('./ipc')
 var resolveMain = require('./resolveMain')
-
+var minimatch = require("minimatch")
```
Comment (gasi) [inline lib/index.js]: Minor: Please move this with the other `require`s, ideally in :abc: order :smile:

File: lib/index.js [hunk 3]
```diff
@@ -118,8 +118,8 @@ function getPrefix(mod) {
   return ~i ? mod.slice(0, i+n.length) : ''
 }
 
-function isPrefixOf(value) {
-  return function (prefix) {
-    return value.indexOf(prefix) === 0
+function isPatternOf(value) {
```
Comment (gasi) [inline lib/index.js]: Let’s rename this to `matches` then, ok?

File: lib/index.js [hunk 4]
```diff
@@ -118,8 +118,8 @@ function getPrefix(mod) {
...
+  return function (ignorePattern) {
+    return minimatch(value,ignorePattern)
```
Comment (gasi) [inline lib/index.js]: Minor: Missing space after comma.

File: test/index.js [hunk 5]
```diff
@@ -76,7 +77,27 @@ test('should not restart the server for ignored modules', function(t) {
...
+        return { exit: t.end.bind(t) }
+      }
+    }
+  })
+})
+
+test('should not restart the server for ignore pattern modules', function(t) {
+  ps = spawn('server.js', function(out) {
+    if (out.match(/touch message.js/)) {
+      setTimeout(touchFile(ignoredPatternFile), 500)
```
Comment (gasi) [inline test/index.js]: This is great :100: :8ball: :tada:

File: lib/index.js (hunk 2 above)
Comment (gasi) [inline lib/index.js]: Also, please use single quotes for strings like the rest of the code.

File: lib/cfg.js (hunk 1 above)
Comment (gasi) [inline lib/cfg.js]: Minor style: Please add spaces around operators such as `==` and `!=` and use their strict equivalents, i.e. `===` and `!==`. Please omit semicolons like the rest of the codebase. Hopefully we can put in a lint tool soon :+1:

**Comment by Casear** [comment]
Hi @gasi , I make some change. Can you verify?

**Comment by gasi** [comment]
@Casear This looks very good. Unfortunately, I am travelling for the next three week and won’t be able to test this. I can test it after I am back or maybe @fgnass can take a look. Thanks again for your contribution :smile:

File: package.json [hunk 6]
```diff
@@ -33,7 +33,7 @@
     "dateformat": "~1.0.4-1.2.3",
     "dynamic-dedupe": "^0.2.0",
     "filewatcher": "~1.1.1",
-    "minimist": "^1.1.3",
```
Comment (dotnil) [inline package.json]: This can't be right. Isn't `minimist` required in `bin/node-dev`

**Comment by gasi** [comment]
@Casear Apologies I hadn’t had a chance to get back to this but due to work reasons, I currently can’t dedicate any time to maintaining `node-dev`. I hope @fgnass or one of the other maintainers will be able to help review your changes :smile:

**Comment by bjornstar** [comment]
I wouldn't mind adding support for pattern matching, but this has gotten outdated.

**Pull request closed without merging**